### PR TITLE
Branch company system

### LINKED ISSUE #54: Add employee hiring action
As a experienced player, I can perform employee hiring action, so that I can develop the virtual startup to.

### PR DESCRIPTION
**Pull request by ZMinghuiZ**
Add company management system that fixes #54

### CODE REVIEW

File: src/main/java/file/Saver.java [hunk 1]
```diff
@@ -36,8 +36,10 @@ private static String constructJson(PlayerProfile playerProfile) {
         return "{\n" +
                 " \"name\": \"" + playerProfile.getName() + "\", \n" +
                 " \"occupation\": \"" + playerProfile.getOccupation() + "\", \n" +
-                " \"asset\": \"" + playerProfile.getAsset().outputAsset() + "\", \n" +
-                " \"health\": \"" + playerProfile.getHealth().outputHealth() + "\", \n" +
+                " \"asset\": \"" + playerProfile.getAsset() + "\", \n" +
+                " \"health\": \"" + playerProfile.getHealth() + "\", \n" +
+                " \"currentRound\": \"" + playerProfile.getCurrentRound() + "\", \n" +
+                " \"isAdvancedPlayer\": \"" + playerProfile.isAdvancedPlayer() + "\" \n" +
                 "}";
```
Comment (ZMinghuiZ) [inline src/main/java/file/Saver.java:43]: How can I store an attribute which is a class?

File: src/main/java/command/CommandFactory.java [hunk 2]
```diff
@@ -23,6 +23,23 @@ public static Command create(String userInput) throws CommandInputException {
...
+        case FIRE:
+            String fireNum = Parser.separateCommand(userInput)[1];
+            return new FireEmployeeCommand(Integer.parseInt(fireNum));
+
+        case ADJUST_SALARY:
+            String[] commandParts = Parser.separateCommand(userInput);
+            String changeType = commandParts[0].toLowerCase();
+            String amount= commandParts[1];
+            return new AdjustSalaryCommand(changeType, Integer.parseInt(amount));
+
```
Comment (ZMinghuiZ) [inline src/main/java/command/CommandFactory.java:42]: would this method be too long? are there any better ways to do the command generation?

File: src/main/java/file/Saver.java (hunk 1 above)
Comment (StargazerX0) [reply src/main/java/file/Saver.java:43]: just convert it into string, we can define a method to do this

File: src/main/java/command/CommandFactory.java (hunk 2 above)
Comment (StargazerX0) [reply src/main/java/command/CommandFactory.java:42]: I think it's fine, we can add other methods to make it shorter

**Review by StargazerX0 — APPROVED** [review]
Well done, I will handle the storage issue

**Pull request merged**